srtcore/access_control.h

```diff
@@ -0,0 +1,81 @@
+#define SRT_REJX_FILEPATH 1003  // The resource type designates a file and the path is either wrong syntax or not found
+#define SRT_REJX_HOSTNOTFOUND 1004 // The `h` host specification was not recognized by the service
+
+// The list of http codes adopted for SRT.
+// An example C++ header for HTTP codes can be found at:
+// https://github.com/j-ulrich/http-status-codes-cpp
+
+// Some of the unused code can be revived in the future, if there
+// happens to be a good reason for it.
+
+#define SRT_REJX_BAD_REQUEST 1400  // General syntax error in the SocketID specification (also a fallback code for undefined cases
+#define SRT_REJX_UNAUTHORIZED 1401  // Authentication failed, provided that the user was correctly identified and access to the required resource would be granted
+#define SRT_REJX_OVERLOAD 1402  // The server is too heavily loaded, or you have exceeded credits for accessing the service and the resource.
+#define SRT_REJX_FORBIDDEN 1403  // Access denied to the resource by any kind of reason.
+#define SRT_REJX_NOTFOUND 1404  // Resource not found at this time.
+#define SRT_REJX_BAD_MODE 1405  // The mode specified in `m` key in StreamID is not supported for this request.
+#define SRT_REJX_UNACCEPTABLE 1406  // The requested parameters specified in SocketID cannot be satisfied for the requested resource. Also when m=publish and the data format is not acceptable.
+// CODE NOT IN USE 407: unused: proxy functionality not predicted
+// CODE NOT IN USE 408: unused: no timeout predicted for listener callback
+#define SRT_REJX_CONFLICT 1409  // The resource being accessed is already locked for modification. This is in case of m=publish and the specified resource is currently read-only.
+// CODE NOT IN USE 410: unused: treated as a specific case of 404
+// CODE NOT IN USE 411: unused: no reason to include lenght in the protocol
+// CODE NOT IN USE 412: unused: preconditions not predicted in AC
+// CODE NOT IN USE 413: unused: AC size is already defined as 512
+// CODE NOT IN USE 414: unused: AC size is already defined as 512
+#define SRT_REJX_NOTSUP_MEDIA 1415  // The media type in not supported by the application. This is the `t` key that specifies the media type as stream, file and auth, possibly extended by the application.
+// CODE NOT IN USE 416: unused: no detailed specification defined
+// CODE NOT IN USE 417: unused: expectations not supported
+// CODE NOT IN USE 418: unused: sharks do not drink tea
+// CODE NOT IN USE 419: not defined in HTTP
+// CODE NOT IN USE 420: not defined in HTTP
+// CODE NOT IN USE 421: unused: misdirection not supported
+// CODE NOT IN USE 422: unused: aligned to general 400
+#define SRT_REJX_LOCKED 1423  // The resource being accessed is locked for any access.
+#define SRT_REJX_FAILED_DEPEND 1424  // The request failed because it specified a dependent session ID that has been disconnected.
```

Comment:
Late found: this could be a duplicate of `SRT_REJX_SESSIONOFF`, so the latter possibly can be removed.